components-core/src/main/java/com/adyen/checkout/components/core/internal/data/api/DefaultAnalyticsRepository.kt

```diff
@@ -22,18 +23,38 @@ class DefaultAnalyticsRepository(
     private val source: AnalyticsSource,
     private val analyticsService: AnalyticsService,
     private val analyticsMapper: AnalyticsMapper,
+    private val clientKey: String,
 ) : AnalyticsRepository {
 
-    override suspend fun sendAnalyticsEvent() {
+    @VisibleForTesting
+    internal var state: State = State.Uninitialized
+        private set
+
+    override suspend fun setupAnalytics() {
+        if (state != State.Uninitialized) return
+        state = State.Initializing
+        Logger.v(TAG, "Setting up analytics")
+
         runSuspendCatching {
-            val queryParameters = analyticsMapper.getQueryParameters(packageName, locale, source)
-            analyticsService.sendEvent(queryParameters)
-            Logger.v(TAG, "Analytics event sent")
+            val analyticsSetupRequest = analyticsMapper.getAnalyticsSetupRequest(packageName, locale, source)
+            analyticsService.setupAnalytics(analyticsSetupRequest, clientKey)
+            state = State.Initialized
+            Logger.v(TAG, "Analytics setup call successful")
+        }.onFailure { e ->
+            state = State.InitializationFailed
+            Logger.e(TAG, "Failed to send analytics setup call", e)
         }
-            .onFailure { e -> Logger.e(TAG, "Failed to send analytics event", e) }
     }
 
     companion object {
         private val TAG = LogUtil.getTag()
     }
+
+    @VisibleForTesting
+    internal sealed class State {
```

Comment:
Names of subclasses are too similar to each other and that kinda makes it hard to read. I think we can rename the parent class to `InitializationState` and sub classes to
`Initializing` -> `InProgress`
`Initialized` -> `Ready`
`InitializationFailed` -> `Failed`

Couldn't find a replacement for `Uninitialized`. We can leave it as is or may be you or @OscarSpruit have an idea?

Reply:
I like your suggestion

Reply:
Fixed 👍